lib/active_merchant/billing/gateways/pay_trace.rb

```diff
@@ -265,15 +265,16 @@ def add_customer_data(post, options)
       end
 
       def add_address(post, creditcard, options)
-        return unless options[:billing_address] || options[:address]
-
-        address = options[:billing_address] || options[:address]
         post[:billing_address] = {}
+
+        if (address = options[:billing_address] || options[:address])
```

Comment:
What happens if the address isn't fully filled out?  Eg. `options[:address][:city]` is `nil`?

Reply:
At least in sandbox nothing happens, the transactions succeeds.